### PR TITLE
Chef RAMsay

### PR DESCRIPTION
**Pull request by jknpj**
chefbot
https://webmshare.com/play/Vn1Wj
**WHAT IS THIS? WHO PUSHED THIS COMMIT? THIS [WIP] CHEF RAMSAY IS RAAAAAAAW!**

This is a port of goon's chefbot. It isn't here to make food for you, it isn't here to make your life easier. It is make here to make you a REAL chef <s>or the annoy the fuck out of you for making shitfood</s>.

It hunts and destroy any food with either shit reagents or over 10 units of corn oil and make your regret your choice of becoming a chef. When emaged it will destroy food WITHOUT shit reagents, taunt the fuck out any live being around him and explode when destroyed, good riddance.

Right now there's no way to acquire him, <s>so i need some ideas about how to make him obtainable.</s> Fuck just keep it as bus only for now.

There's a kick_act to turn it off but be caferul to not break it in the process.

### CODE REVIEW

**Comment by Shadowmech88** [comment]
You could make him have a % chance to spawn when a burned mess is spawned.

File: goon/code/obj/machinery/bot/chefbot.dm [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+	desc = "(icon, name, concept, and any kind of consistency or sense is currently pending)"
+	icon = 'goon/icons/obj/aibots.dmi'
+	icon_state = "chefbot-idle"
+	density = 1
+	anchored = 0
+	on = 1 // ACTION
+	health = 50
+	var/raging = 0
+	var/list/calledout = list()
+	var/shit_reagents = list(TOXIN,STOXIN,SLIMEJELLY,PLASMA,NANITES,HONKSERUM,SILENCER,DISCOUNT,AMATOXIN,PSILOCYBIN,CARPPHEROMONES,BUSTANUT,ROGAN,MOONROCKS,TOXICWASTE,CHEMICAL_WASTE,HORSEMEAT,OFFCOLORCHEESE,BONEMARROW,IRRADIATEDBEANS,MUTATEDBEANS,MINTTOXIN,MERCURY,MINDBREAKER,SPIRITBREAKER,RADIUM,URANIUM,CARPOTOXIN,ZOMBIEPOWDER,AMUTATIONTOXIN,PACID,CHLORALHYDRATE,LITHIUM,HELL_RAMEN)
```
Comment (Exxion) [inline goon/code/obj/machinery/bot/chefbot.dm]: Why no Silencer?
And I know why not minttoxin, but personally I think that should be allowed

**Comment by Exxion** [comment]
He just like
Bursts out of the wall in a burning rage?
Fund it

File: goon/code/obj/machinery/bot/chefbot.dm (hunk 1 above)
Comment (Shadowmech88) [reply goon/code/obj/machinery/bot/chefbot.dm]: Players can't even cook anything with mint toxin, can they?
RAMsay would just smash the mint in the backroom every time he's spawned.

**Comment by Ephx1** [comment]
Bursts out of the _Microwave_ in burning rage.

**Comment by ghost** [comment]
Make him buildable with a proximity sensor, a couple forks, a chef hat, and a box. Maybe minus the chef hat if that seems like too much.

File: goon/code/obj/machinery/bot/chefbot.dm (hunk 1 above)
Comment (jknpj) [reply goon/code/obj/machinery/bot/chefbot.dm]: Being forced to stay silent after eating shitfood should not be torelable.

Minttoxin seems reasonable, as you start with it in your backroom, but it is still pretty much lethal.

**Comment by jknpj** [comment]
@unusualcrow the thing makes buttbots look like child's play in the annoying scale, you don't want them flooding the station

**Comment by RFultra** [comment]
no ghandi's flip flop quote?

**Comment by eplgr** [comment]
This is a feature that will guarantee the food is not harmful, right? With no effort being made.
Please take into account that fact.

File: goon/code/obj/machinery/bot/chefbot.dm (hunk 1 above)
Comment (Exxion) [reply goon/code/obj/machinery/bot/chefbot.dm]: I
Don't think Silencer actually mutes you

Comment (jknpj) [reply goon/code/obj/machinery/bot/chefbot.dm]: `/datum/reagent/drink/silencer/on_mob_life(var/mob/living/M)

	if(..())
		return 1
	M.silent = max(M.silent, 15)`

**Comment by Probe1** [comment]
DNM is for the wip title obviously but just throwing that out there.

**Comment by Ephx1** [comment]
Oh and about the way to get him. If you do go for something silly like him just appearing from shit food, have it be from a fully upgraded microwave. Or craftable with a microwave, two arms, speaker, chef's hat, prox sensor, and knife.

**Comment by jknpj** [comment]
thesalid

**Comment by Ephx1** [comment]
That's the delicious undercooked irony.

**Comment by Kurfursten** [comment]
Pity this is probably going to hit before I can rewrite nutrition. It would be nice if Chef Ramsay would keep track of your cooking and berate you for too much fat / sweets / cholesterol.

**Comment by jknpj** [comment]
We can rebuild him, we have the technology

File: goon/code/obj/machinery/bot/chefbot.dm [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+			raging = 1
+			update_icon()
+			for(var/mob/living/carbon/human/M in view(7, src))
+				if(M.mind)
+					if(M.mind.assigned_role == "Chef")
+						thechef = M
+						is_thechef_the_chef = 1
+						break
+				if(M.wear_id)
+					if(findtext(M.wear_id:assignment, "chef") || findtext(M.wear_id:assignment, "cook"))
```
Comment (PJB3005) [inline goon/code/obj/machinery/bot/chefbot.dm]: IT'S THE RUNTIME OPERATOR EVERYBODY RUN!

File: goon/code/obj/machinery/bot/chefbot.dm [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+						thechef = M
+						is_thechef_the_chef = 1
+						break
+				if(!thechef)
+					thechef = M
+				if(!dork)
+					if(M.client)
+						if(M.client.IsByondMember())
```
Comment (PJB3005) [inline goon/code/obj/machinery/bot/chefbot.dm:83]: Not sure whether this is hilarious or we should give no incentive to support Byond.

File: goon/code/obj/machinery/bot/chefbot.dm [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+		return 1
+	return 0
+
+/obj/machinery/bot/chefbot/attackby(obj/item/W as obj, mob/user as mob)
+	if(istype(W, /obj/item/weapon/card/emag))
+		emag_act(user, W)
+	else
+		src.visible_message("<span class = 'warning'>[user] hits [src] with [W]!</span>")
+		switch(W.damtype) // SEE THIS DAMAGE? IT'S RAAAAAAAAAAW!
+			if("fire")
```
Comment (PJB3005) [inline goon/code/obj/machinery/bot/chefbot.dm]: Aren't there defines for this?

File: goon/code/obj/machinery/bot/chefbot.dm [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+				src.health -= W.force * 0.5
+			if("brute")
+				src.health -= W.force * 0.5
+			else
```
Comment (PJB3005) [inline goon/code/obj/machinery/bot/chefbot.dm]: Wha

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: goon/code/obj/machinery/bot/chefbot.dm (hunk 3 above)
Comment (Kurfursten) [reply goon/code/obj/machinery/bot/chefbot.dm:83]: well, it's not really an incentive, the only thing that it is used for is to determine if he can ask if they "bought their [cooking] license"

seems harmless

**Comment by carshalash** [comment]
Those roburger possibilities are fucking great

**Comment by kol1th** [comment]
muh perspective

**Comment by ghost** [comment]
@jknpj https://my.mixtape.moe/nnnaot.ogg
Use this for the dramatic sound. 
There was a bit of a "tss" sound left over at the beginning from someone talking, so I cut that out and I turned up the volume a little since it was fairly quiet.

File: goon/code/obj/machinery/bot/chefbot.dm [hunk 6]
```diff
@@ -93,7 +97,7 @@
 					say("WHO COOKED THIS SHIT?")
 				sleep(2 SECONDS)
 				if(shitfood) // fix for cannot read null.name (the food sometimes no longer exists after a sleep (because people eat it I assume)) - haine
-					say("THIS [uppertext(shitfood.name)] [why_is_it_bad()][dork? ", DID YOU BUY YOUR LICENSE, [uppertext(dork)]?" : "!"]")
+					say("THIS [uppertext(shitfood.name)] [why_is_it_bad()][dork? ", DID YOU BUY YOUR FUCING COOKING LICENSE, [uppertext(dork)]?" : "!"]")
```
Comment (ghost) [inline goon/code/obj/machinery/bot/chefbot.dm]: fucing

**Review by PJB3005 — APPROVED** [review]

File: goon/code/obj/machinery/bot/chefbot.dm [hunk 7]
```diff
@@ -0,0 +1,193 @@
+/obj/machinery/bot/chefbot
+	name = "Chef RAMsay"
+	desc = "(icon, name, concept, and any kind of consistency or sense is currently pending)"
```
Comment (Probe1) [inline goon/code/obj/machinery/bot/chefbot.dm]: Whoops forgot something

File: goon/code/obj/machinery/bot/chefbot.dm [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+	icon = 'goon/icons/obj/aibots.dmi'
+	icon_state = "chefbot-idle"
+	density = 1
+	anchored = 0
+	on = 1 // ACTION
```
Comment (Probe1) [inline goon/code/obj/machinery/bot/chefbot.dm:8]: There's no way to turn it off.

File: goon/code/obj/machinery/bot/chefbot.dm [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+			to_chat(user, "<span class = 'warning'>You short out the restraining bolt on [src].</span>")
+			drama()
+			sleep(2 SECONDS)
+			say("FORECAST FOR TOMORROW? 100% CHANCE OF TEARS.")
+		sleep(2 SECONDS)
+		update_icon()
+		return 1
+	return 0
+
+/obj/machinery/bot/chefbot/attackby(obj/item/W as obj, mob/user as mob)
```
Comment (Probe1) [inline goon/code/obj/machinery/bot/chefbot.dm:166]: There's no way to punch it but there's no attack delay on hitting it with anything else.  Don't worry though!  MELEE ATTACKING IT DOES NO DAMAGE.

**Review by Probe1 — CHANGES_REQUESTED** [review]
Fix it ignoring melee damage and the description.

Also consider having it bark out lines at random because unless it happens upon dodgy food it does nothing of interest.  Having a random bark() at passers would be good.

File: goon/code/obj/machinery/bot/chefbot.dm (hunk 9 above)
Comment (Probe1) [reply goon/code/obj/machinery/bot/chefbot.dm:166]: Also you can kick it and that DOES do damage.

Comment (Kurfursten) [reply goon/code/obj/machinery/bot/chefbot.dm:166]: can you punch any bot? usually that opens a menu

Comment (Probe1) [reply goon/code/obj/machinery/bot/chefbot.dm:166]: Hopefully but probably not.  Still needs fixed, especially the complete lack of code handling being attacked by an object that isn't a projectile/beam/leg.

File: goon/code/obj/machinery/bot/chefbot.dm (hunk 8 above)
Comment (Shadowmech88) [reply goon/code/obj/machinery/bot/chefbot.dm:8]: Possibly intentional?

Comment (jknpj) [reply goon/code/obj/machinery/bot/chefbot.dm:8]: It was intentional but i ended adding a kick_act to turn it off.

File: goon/code/obj/machinery/bot/chefbot.dm [hunk 10]
```diff
@@ -97,7 +97,7 @@
 					say("WHO COOKED THIS SHIT?")
 				sleep(2 SECONDS)
 				if(shitfood) // fix for cannot read null.name (the food sometimes no longer exists after a sleep (because people eat it I assume)) - haine
-					say("THIS [uppertext(shitfood.name)] [why_is_it_bad()][dork? ", DID YOU BUY YOUR FUCKING COOKING LICENSE, [uppertext(dork)]?" : "!"]")
+					say("THIS [uppertext(shitfood.name)] [why_is_it_bad()][dork? ", DID YOU BUY YOUR FUCING COOKING LICENSE, [uppertext(dork)]?" : "!"]")
```
Comment (ghost) [inline goon/code/obj/machinery/bot/chefbot.dm]: you did it again you DOUBLE SHITCODER

Comment (Probe1) [inline goon/code/obj/machinery/bot/chefbot.dm]: i dont get it

**Review by Probe1 — APPROVED** [review]

File: goon/code/obj/machinery/bot/chefbot.dm (hunk 10 above)
Comment (Shadowmech88) [reply goon/code/obj/machinery/bot/chefbot.dm]: This is probably a typo and should be fixed.

Comment (jknpj) [reply goon/code/obj/machinery/bot/chefbot.dm]: jesus christ

Comment (jknpj) [reply goon/code/obj/machinery/bot/chefbot.dm]: yeah i'm using an old version of the file, haha

**Comment by Shadowmech88** [comment]
Should there have been a changelog for this?

**Comment by Probe1** [comment]
Yeah there should have been but that's on me.

**Pull request merged**